main.py

```diff
@@ -212,23 +235,17 @@ def threadLoop(self):
                             lastLon = lon
                             lmu = time.time()
 
-                with self.lock:
-                    self.lastLatitude = lastLat
-                    self.lastLongitude = lastLon
-                    self.lastMapUpdate = lmu
-                    lgps = self.lastgps
+                lastLatitude = lastLat
+                lastLongitude = lastLon
+                lastMapUpdate = lmu
+                lgps = lastgps
 
-                if time.time() - lgps >= 1:
-                    self.updateMark(self.latitude, self.longitude)
+                if time.time() - lgps >= 1:  # TODO sort out what this code does and if it can handle change to way it uses data
+                    with self.lock:
+                        self.updateMark(self.latitude, self.longitude)
```

Comment:
What exactly needs to be sorted out here?

Reply:
I believe that was a personal note/reminder from @DSep's commit. I kind of based my changes off of his working branch before he finished his work. I was trying to minimize how many conflicts I'd create in his branch because his work is closely related.

Reply:
You don't need to sort anything out dw.

Reply:
Sounds good.

Reply:
Yeah, it was a note to self to sort out how data was handled here.